clients/socket-client.rst

```diff
@@ -63,4 +63,10 @@ use case would be::
     ];
     $client = new Client($messageFactory, $options);
 
+.. warning::
+
+    This client assume that the request is compliant with HTTP 2.0, 1.1 or 1.0 standard. So a request without a ``Host`` header, or
+    with a body but without a ``Content-Length`` will certainly fail. You can make sure that requests used by this client are valid, by
+    decorating it with :doc:`plugins </plugins/introduction>` like ``ContentLengthPlugin`` or ``HeaderDefaultsPlugin``.
```

Comment:
by using the plugins ContentLengthPlugin and HeaderDefaultsPlugin.

i think we need to think how to make it more clear that the two things are both required. maybe first talk about the host and HeaderDefaultsPlugin, then talk about the body and content length?